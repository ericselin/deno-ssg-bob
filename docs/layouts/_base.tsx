/** @jsx h */

import { Component, getPath, h, readContents } from "../../mod.ts";

const baseCss = getPath(import.meta.url, "_base.css");

export const Base: Component = async (
  { children },
  { needsCss },
) => (
  <html lang="en">
    <head>
      <meta charSet="UTF-8" />
      <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <meta http-equiv="X-UA-Compatible" content="ie=edge" />
      <title>bob the static site builder</title>
      <link rel="preconnect" href="https://fonts.gstatic.com" />
      <link
        href="https://fonts.googleapis.com/css2?family=Inter:wght@400;700&family=Source+Code+Pro:wght@400;700;900&display=swap"
        rel="stylesheet"
      />
      <style>
        {await readContents([baseCss, ...needsCss])}
      </style>
    </head>

    <body>
      <header>
        <a href="/">bob</a>
        <ul>
          <li>
            <a href="/getting-started/">Getting started</a>
          </li>
          <li>
            <a href="/scenarios/">Common scenarios</a>
          </li>
        </ul>
      </header>
      {children}
      <footer>
        Static sites should be fast, including updating content
      </footer>
    </body>
  </html>
);
