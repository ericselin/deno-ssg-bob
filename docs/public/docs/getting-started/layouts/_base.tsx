/** @jsx h */

import { Component, h } from "../../../../../mod.ts";

export const Base: Component = (props) => (
  <html>
    <head>
      <title>This is my new site!</title>
    </head>
    <body>
      <header>
        Main header
      </header>
      {props.children}
    </body>
  </html>
);
