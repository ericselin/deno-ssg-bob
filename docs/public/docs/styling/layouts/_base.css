body {
  font-family: serif;
  margin: 1rem auto;
  max-width: 50rem;
}
