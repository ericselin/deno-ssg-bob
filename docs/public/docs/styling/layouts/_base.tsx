/** @jsx h */

import { Component, h, readContents } from "../../../../../mod.ts";

export const Base: Component = async (
  { children },
  { needsCss },
) => (
  <html lang="en">
    <head>
      <title>This is my page</title>
      <style>
        {await readContents(needsCss)}
      </style>
    </head>
    <body>
      {children}
    </body>
  </html>
);

// Tip! You can add CSS files to any component, also this one
Base.needsCss = "_base.css";
