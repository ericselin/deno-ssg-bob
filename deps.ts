/*
Copyright 2021 Eric Selin

This file is part of `bob`.

`bob` is free software: you can redistribute it and/or modify
it under the terms of the GNU Lesser General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

`bob` is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
GNU Lesser General Public License for more details.

You should have received a copy of the GNU Lesser General Public License
along with `bob`. If not, see <https://www.gnu.org/licenses/>.

Please contact the developers via GitHub <https://www.github.com/ericselin>
or email eric.selin@gmail.com <mailto:eric.selin@gmail.com>
*/

export {
  assert,
  assertEquals,
} from "https://deno.land/std@0.109.0/testing/asserts.ts";
export * as path from "https://deno.land/std@0.109.0/path/mod.ts";
export * as yaml from "https://deno.land/std@0.109.0/encoding/yaml.ts";
export { exists } from "https://deno.land/std@0.109.0/fs/exists.ts";
export { walk } from "https://deno.land/std@0.109.0/fs/walk.ts";
export { expandGlob } from "https://deno.land/std@0.109.0/fs/expand_glob.ts";
export type { WalkEntry } from "https://deno.land/std@0.109.0/fs/walk.ts";
export { Marked as md } from "https://deno.land/x/markdown@v2.0.0/mod.ts";
export { parse as parseFlags } from "https://deno.land/std@0.110.0/flags/mod.ts";

import * as log from "https://deno.land/std@0.109.0/log/mod.ts";
export { log };
export type Logger = typeof log;

export { serve } from "https://deno.land/std@0.121.0/http/server.ts";
