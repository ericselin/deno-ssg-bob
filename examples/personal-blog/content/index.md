---
modified: 2021-04-28T23:22:57+02:00
---

I like to code. And I like to develop new things. Most of all, I'm an
entrepreneur at heart.

### Coding

Elegant code is the dream. Simple solutions are the best.

### Entrepreneurship

Test early and often because it's fun and effective.

## Projects

### Finnish Baby Box

This is my current day job. I manage the technical implementation of our
custom-built ecommerce solution. Official title: co-founder and CTO.

[finnishbabybox.com](https://www.finnishbabybox.com)

### Humla

I tried every todo app out there and finally decided to make my own that
works the way I want. Active users: 3.

[humla.app](https://humla.app)