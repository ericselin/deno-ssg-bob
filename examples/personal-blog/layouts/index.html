{{ define "main" }}

<section class="hero">
  <h1>Hi, I'm <span>Eric</span></h1>
  <h2>I'm a developer.</h2>
</section>

<main>
  {{ .Content }}
</main>

{{ end }}
