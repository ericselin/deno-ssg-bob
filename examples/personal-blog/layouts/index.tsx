/** @jsx h */

import { Component, h } from "./deps.ts";
import { Base } from "./_base.tsx";

const Index: Component = (
  _props,
  {page: {content}}
) => (
  <Base>
    <section class="hero">
      <h1>
        Hi, I'm <span>Eric</span>
      </h1>
      <h2>I'm a developer.</h2>
    </section>

    <main>
      {content}
    </main>
  </Base>
);

Index.needsCss = "index.css";

export default Index;
