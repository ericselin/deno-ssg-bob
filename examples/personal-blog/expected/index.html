<html lang="en"><head><meta charSet="UTF-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta http-equiv="X-UA-Compatible" content="ie=edge"><title>I'm Eric, and I'm a developer | ericselin.dev</title><link rel="preconnect" href="https://fonts.gstatic.com"><link href="https://fonts.googleapis.com/css2?family=Inter:wght@400;700&family=Source+Code+Pro:wght@400;700;900&display=swap" rel="stylesheet"><style>.hero {
  font-family: var(--font-heading);
  max-width: var(--max-width);
  margin: calc(3 * var(--margin)) auto;
}

.hero h1 {
  font-size: 4rem;
}

.hero h1, .hero h2 {
  margin: 0;
}

.hero h1 span {
  position: relative;
  z-index: 1;
}

.hero h1 span::before {
  content: ' ';
  display: block;
  position: absolute;
  height: 1rem;
  bottom: -0.25rem;
  background: var(--color-accent);
  left: 0;
  right: 0;
  z-index: -1;
}

.hero h2 {
  color: inherit;
  font-weight: 400;
  margin-top: 1rem;
}

:root {
  --nord0: #2e3440;
  --nord1: #3b4252;
  --nord2: #434c5e;
  --nord3: #4c566a;
  --nord4: #d8dee9;
  --nord5: #e5e9f0;
  --nord6: #eceff4;
  --nord7: #8fbcbb;
  --nord8: #88c0d0;
  --nord9: #81a1c1;
  --nord10: #5e81ac;
  --nord11: #bf616a;
  --nord12: #d08770;
  --nord13: #ebcb8b;
  --nord14: #a3be8c;
  --nord15: #b48ead;

  --margin: 1rem;
  --max-width: min(600px, 100% - 4 * var(--margin));

  --color-bg: var(--nord0);
  --color-fg: var(--nord6);
  --color-body: var(--nord1);
  --color-indent: var(--nord3);
  --color-muted: var(--nord4);
  --color-accent: var(--nord8);

  --font-text: 'Inter', sans-serif;
  --font-heading: 'Source Code Pro', monospace;
  --font-mono: 'Source Code Pro', monospace;
}

@media (min-width: 500px) {
  :root {
    --margin: 1.5rem;
  }
}

@media (min-width: 768px) {
  :root {
    --margin: 2rem;
  }
}

body {
  padding: 0;
  margin: 0;
  font-family: var(--font-text);
  background: var(--color-body);
  color: var(--color-fg);
}

body > header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: var(--margin);
  color: var(--color-muted);
  font-family: var(--font-heading);
}

body > header > a {
  text-decoration: none;
  font-weight: 900;
  font-size: 2rem;
  background: var(--color-accent);
  width: 2rem;
  height: 2rem;
  line-height: 1.5rem;
  text-align: center;
  border-radius: 25%;
  color: var(--color-body);
}

body > header > a:hover {
  color: var(--color-bg);
}

body > header ul {
  display: grid;
  grid-auto-flow: column;
  gap: 1rem;
  list-style: none;
  padding: 0;
  margin: 0;
}

body > h1 {
  max-width: var(--max-width);
  margin: calc(2 * var(--margin)) auto;
  padding: 0 var(--margin);
  color: var(--color-fg);
  font-family: var(--font-heading);
}

main {
  max-width: var(--max-width);
  margin: var(--margin) auto;
  padding: 0 var(--margin);
  background: var(--color-bg);
  border-radius: 1rem;
  line-height: 1.25;
}

main::before,
main::after {
  content: ' ';
  display: block;
  contain: layout;
  margin: var(--margin) 0;
}

h1 a,
h2 a,
h3 a,
h4 a,
h5 a,
h6 a {
  text-decoration: none;
}

a {
  color: inherit;
  transition: color ease-in 200ms;
}

a:hover {
  color: var(--color-accent);
}

pre {
  max-width: 100%;
  display: block;
  overflow: auto;
}

code {
  font-family: var(--font-mono);
  line-height: initial;
}

article + article {
  margin-top: calc(2 * var(--margin));
}

article + article::before {
  content: '';
  display: block;
  height: 1px;
  width: 50%;
  background: var(--color-indent);
  margin: 0 auto calc(2 * var(--margin));
}

img {
  display: block;
  max-width: 100%;
}

.button-link {
  border: 1px solid var(--color-accent);
  border-radius: 0.5rem;
  padding: 0.5rem 1rem;
  display: inline-block;
  text-decoration: none;
}

body > footer {
  margin: var(--margin);
  text-align: center;
  color: var(--color-muted);
  font-family: var(--font-heading);
}
</style></head><body><header><a href="/">e</a><ul><li><a href="/blog/">Blog</a></li><li><a href="/experiments/">Experiments</a></li></ul></header><section class="hero"><h1>Hi, I'm <span>Eric</span></h1><h2>I'm a developer.</h2></section><main><p>I like to code. And I like to develop new things. Most of all, I&#39;m an
entrepreneur at heart.</p>
<h3 id="coding">Coding</h3>
<p>Elegant code is the dream. Simple solutions are the best.</p>
<h3 id="entrepreneurship">Entrepreneurship</h3>
<p>Test early and often because it&#39;s fun and effective.</p>
<h2 id="projects">Projects</h2>
<h3 id="finnish-baby-box">Finnish Baby Box</h3>
<p>This is my current day job. I manage the technical implementation of our
custom-built ecommerce solution. Official title: co-founder and CTO.</p>
<p><a href="https://www.finnishbabybox.com">finnishbabybox.com</a></p>
<h3 id="humla">Humla</h3>
<p>I tried every todo app out there and finally decided to make my own that
works the way I want. Active users: 3.</p>
<p><a href="https://humla.app">humla.app</a></p>
</main><footer>There's no such thing as the cloud - it's just someone else's computer.</footer></body></html>